### ISSUE TITLE
IDE freeze when opening log file

### ISSUE THREAD

**Issue by CrazyCoder** [JetBrains/ideolog#177]
```
"AWT-EventQueue-0" prio=0 tid=0x0 nid=0x0 blocked
     java.lang.Thread.State: BLOCKED
 on java.lang.Object@5f5c2 owned by "Timer-2" Id=278
	at com.intellij.ideolog.util.IdeologDocumentContext.getEvent(IdeologDocumentContext.kt:96)
	at com.intellij.ideolog.highlighting.LogHighlightingIterator.<init>(LogHighlightingIterator.kt:51)
	at com.intellij.ideolog.highlighting.LogEditorHighlighter.createIterator(LogFileHighlighterFactory.kt:45)
	at com.intellij.openapi.editor.impl.view.IterationState.<init>(IterationState.java:130)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paintLineFragments(EditorPainter.java:1474)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paintBackground(EditorPainter.java:361)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paint(EditorPainter.java:190)
	at com.intellij.openapi.editor.impl.view.EditorPainter.paint(EditorPainter.java:72)
	at com.intellij.openapi.editor.impl.view.EditorView.paint(EditorView.java:282)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$paint$48(EditorImpl.java:2007)
	at com.intellij.openapi.editor.impl.EditorImpl$$Lambda$3333/0x0000000801ab2b88.run(Unknown Source)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.ReadAction$$Lambda$1849/0x0000000801223b30.compute(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:909)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:65)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:52)
	at com.intellij.openapi.editor.impl.EditorImpl.paint(EditorImpl.java:1989)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.lambda$paintComponent$1(EditorComponentImpl.java:277)
	at com.intellij.openapi.editor.impl.EditorComponentImpl$$Lambda$3332/0x0000000801ab1cd8.run(Unknown Source)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.ReadAction$$Lambda$1849/0x0000000801223b30.compute(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:909)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:65)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:52)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paintComponent(EditorComponentImpl.java:265)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1115)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paint(EditorComponentImpl.java:143)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JViewport.paint(JViewport.java:736)
	at com.intellij.ui.components.JBViewport.paint(JBViewport.java:235)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at com.intellij.openapi.fileEditor.impl.EditorTabs.paintChildren(EditorTabbedContainer.kt:613)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5312)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(BufferStrategyPaintManager.java:247)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1347)
	at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5260)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5070)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:882)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:865)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:865)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:838)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:787)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1909)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue$$Lambda$766/0x000000080079d048.run(Unknown Source)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueueKt$$Lambda$765/0x000000080079c748.run(Unknown Source)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:114)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$$Lambda$764/0x0000000800796850.run(Unknown Source)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:386)
	at com.intellij.openapi.progress.impl.PlatformTaskSupportKt.pumpEventsForHierarchy(PlatformTaskSupport.kt:396)
	at com.intellij.openapi.progress.impl.PlatformTaskSupportKt.access$pumpEventsForHierarchy(PlatformTaskSupport.kt:1)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$2.invoke(PlatformTaskSupport.kt:156)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$2.invoke(PlatformTaskSupport.kt:134)
	at com.intellij.openapi.application.impl.ModalityKt.inModalContext(modality.kt:12)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:134)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.access$runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:48)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$1.invoke(PlatformTaskSupport.kt:122)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$1.invoke(PlatformTaskSupport.kt:118)
	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:86)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:118)
	at com.intellij.platform.ide.progress.TasksKt.runWithModalProgressBlocking(tasks.kt:173)
	at com.intellij.platform.ide.progress.TasksKt.runWithModalProgressBlocking(tasks.kt:100)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openEditorImpl(FileEditorManagerImpl.kt:1346)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileEditor(FileEditorManagerImpl.kt:1312)
	at com.intellij.openapi.fileEditor.FileNavigatorImplKt.navigateInAnyFileEditor(FileNavigatorImpl.kt:97)
	at com.intellij.openapi.fileEditor.FileNavigatorImplKt.access$navigateInAnyFileEditor(FileNavigatorImpl.kt:1)
	at com.intellij.openapi.fileEditor.FileNavigatorImpl.navigateInEditor(FileNavigatorImpl.kt:55)
	at com.intellij.openapi.fileEditor.FileNavigatorImpl.navigateInEditorOrNativeApp(FileNavigatorImpl.kt:50)
	at com.intellij.openapi.fileEditor.FileNavigatorImpl.navigate(FileNavigatorImpl.kt:32)
	at com.intellij.openapi.fileEditor.OpenFileDescriptor.navigate(OpenFileDescriptor.java:88)
	at com.intellij.psi.impl.source.PsiFileImpl.navigate(PsiFileImpl.java:907)
	at com.intellij.ide.projectView.impl.nodes.AbstractPsiBasedNode.navigate(AbstractPsiBasedNode.java:309)
	at com.intellij.ide.projectView.impl.nodes.AbstractPsiBasedNode.navigate(AbstractPsiBasedNode.java:316)
	at com.intellij.ide.projectView.impl.nodes.PsiFileNode.navigate(PsiFileNode.java:117)
	at com.intellij.platform.ide.navigation.impl.IdeNavigationServiceKt$navigateToSource$2$1.invoke(IdeNavigationService.kt:141)
	at com.intellij.platform.ide.navigation.impl.IdeNavigationServiceKt$navigateToSource$2$1.invoke(IdeNavigationService.kt:140)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:321)
	at com.intellij.openapi.progress.CoroutinesKt.access$blockingContextInner(coroutines.kt:1)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invokeSuspend(coroutines.kt:198)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:197)
	at com.intellij.platform.ide.navigation.impl.IdeNavigationServiceKt$navigateToSource$2.invokeSuspend(IdeNavigationService.kt:140)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at com.intellij.openapi.application.impl.DispatchedRunnable.run(DispatchedRunnable.kt:43)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:843)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:463)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:75)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1436)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:124)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:44)
	at com.intellij.openapi.application.impl.FlushQueue$$Lambda$398/0x0000000800565ba8.run(Unknown Source)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue$$Lambda$766/0x000000080079d048.run(Unknown Source)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueueKt$$Lambda$765/0x000000080079c748.run(Unknown Source)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:114)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$$Lambda$764/0x0000000800796850.run(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:843)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:391)
	at com.intellij.openapi.progress.impl.PlatformTaskSupportKt.pumpEventsForHierarchy(PlatformTaskSupport.kt:396)
	at com.intellij.openapi.progress.impl.PlatformTaskSupportKt.access$pumpEventsForHierarchy(PlatformTaskSupport.kt:1)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$2.invoke(PlatformTaskSupport.kt:156)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$2.invoke(PlatformTaskSupport.kt:134)
	at com.intellij.openapi.application.impl.ModalityKt.inModalContext(modality.kt:12)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:134)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.access$runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:48)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$1.invoke(PlatformTaskSupport.kt:122)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport$runWithModalProgressBlockingInternal$1.invoke(PlatformTaskSupport.kt:118)
	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:86)
	at com.intellij.openapi.progress.impl.PlatformTaskSupport.runWithModalProgressBlockingInternal(PlatformTaskSupport.kt:118)
	at com.intellij.platform.ide.progress.TasksKt.runWithModalProgressBlocking(tasks.kt:173)
	at com.intellij.platform.ide.progress.TasksKt.runWithModalProgressBlocking(tasks.kt:100)
	at com.intellij.util.OpenSourceUtilKt.navigateToSource(OpenSourceUtil.kt:49)
	at com.intellij.util.OpenSourceUtil.navigateToSource(OpenSourceUtil.java:150)
	at com.intellij.ui.AutoScrollToSourceTaskManagerImpl$scheduleScrollToSource$1.invokeSuspend(AutoScrollToSourceTaskManagerImpl.kt:49)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at com.intellij.openapi.application.impl.DispatchedRunnable.run(DispatchedRunnable.kt:43)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:843)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:463)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:75)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1436)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:124)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:44)
	at com.intellij.openapi.application.impl.FlushQueue$$Lambda$398/0x0000000800565ba8.run(Unknown Source)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base@17.0.9/java.security.AccessController.executePrivileged(AccessController.java:776)
	at java.base@17.0.9/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base@17.0.9/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue$$Lambda$766/0x000000080079d048.run(Unknown Source)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueueKt$$Lambda$765/0x000000080079c748.run(Unknown Source)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$$Lambda$764/0x0000000800796850.run(Unknown Source)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:848)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:391)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

"Timer-2" prio=0 tid=0x0 nid=0x0 runnable
     java.lang.Thread.State: RUNNABLE

	at java.base@17.0.9/java.util.regex.Pattern$BmpCharProperty.match(Pattern.java:3954)
	at java.base@17.0.9/java.util.regex.Pattern$Branch.match(Pattern.java:4734)
	at java.base@17.0.9/java.util.regex.Pattern$GroupHead.match(Pattern.java:4789)
	at java.base@17.0.9/java.util.regex.Pattern$GroupTail.match(Pattern.java:4820)
	at java.base@17.0.9/java.util.regex.Pattern$BmpCharPropertyGreedy.match(Pattern.java:4329)
	at java.base@17.0.9/java.util.regex.Pattern$BmpCharPropertyGreedy.match(Pattern.java:4329)
	at java.base@17.0.9/java.util.regex.Pattern$BmpCharPropertyGreedy.match(Pattern.java:4329)
	at java.base@17.0.9/java.util.regex.Pattern$BmpCharPropertyGreedy.match(Pattern.java:4329)
	at java.base@17.0.9/java.util.regex.Pattern$BmpCharPropertyGreedy.match(Pattern.java:4329)
	at java.base@17.0.9/java.util.regex.Pattern$GroupHead.match(Pattern.java:4789)
	at java.base@17.0.9/java.util.regex.Pattern$GroupTail.match(Pattern.java:4820)
	at java.base@17.0.9/java.util.regex.Pattern$CharPropertyGreedy.match(Pattern.java:4290)
	at java.base@17.0.9/java.util.regex.Pattern$GroupHead.match(Pattern.java:4789)
	at java.base@17.0.9/java.util.regex.Pattern$Begin.match(Pattern.java:3672)
	at java.base@17.0.9/java.util.regex.Matcher.search(Matcher.java:1728)
	at java.base@17.0.9/java.util.regex.Matcher.find(Matcher.java:745)
	at com.intellij.ideolog.util.IdeologDocumentContext.detectLogFileFormat(IdeologDocumentContext.kt:83)
	at com.intellij.ideolog.util.IdeologDocumentContext.getEventEndLine(IdeologDocumentContext.kt:110)
	at com.intellij.ideolog.util.IdeologDocumentContext.getEvent(IdeologDocumentContext.kt:98)
	at com.intellij.ideolog.highlighting.LogParsingUtils.getEvent(LogParsingUtils.kt:80)
	at com.intellij.ideolog.highlighting.LogParsingUtils.getEvent(LogParsingUtils.kt:74)
	at com.intellij.ideolog.highlighting.LogEvent$Companion.fromEditor(LogEvent.kt:69)
	at com.intellij.ideolog.highlighting.LogFileMapRenderer$initEventMaps$$inlined$scheduleAtFixedRate$1.run(Timer.kt:193)
	at java.base@17.0.9/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base@17.0.9/java.util.TimerThread.run(Timer.java:516)

```

(ZenDesk support ticket 5972952).

**Comment by mmihalev** [comment]
Please, note that this issue can be observed only when Ideolog plugin is active! If plugin is disabled, everything works without any problems.

**Comment by boot-vue** [comment]
same problem

**Comment by sergyu91ma** [comment]
same

**Comment by NeiroNext** [comment]
Same problem. Turned off  this plugin -.-

**Comment by zhsama** [comment]
same

**Comment by HBat11233** [comment]
version 222.2.0.0 can used

**Comment by nbatzak** [comment]
I'm encountering the same issue. Upon deactivating the plugin, the problem ceased. Is there a way for me to utilize a previous version?

**Comment by HBat11233** [comment]
https://plugins.jetbrains.com/plugin/9746-ideolog/versions

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: nbatzak ***@***.***>
发送时间: Saturday, February 3, 2024 4:41:03 PM
收件人: JetBrains/ideolog ***@***.***>
抄送: HBat11233 ***@***.***>; Comment ***@***.***>
主题: Re: [JetBrains/ideolog] IDE freeze when opening log file (Issue #177)


I'm encountering the same issue. Upon deactivating the plugin, the problem ceased. Is there a way for me to utilize a previous version?

―

**Comment by matpag** [comment]
Same problem for me.
It seems like it tries to parse all the stacktrace path to reference the code and enable click-opening on them.

If you have a big log file IntelliJ completely freezes and the only way to proceed is force killing the process and restart.

I had to change the extension to all my logs from `any_log_name.log` to `any_log_name.txt`, in this way IJ does not try to parse everything

**Comment by KipchirchirIan** [comment]
@matpag Just use an older version like the one suggested by @HBat11233 until a fix is provided. Tested it and it still works well. No need to change anything else or disable it.

**Comment by CrazyCoder** [comment]
Another report: https://youtrack.jetbrains.com/issue/IDEA-345300/Find-in-files-hangs-with-high-cpu.

**Comment by winvelod** [comment]
Some problem.
Version: 222.3.1.0

**Comment by twelife** [comment]
Same problem.
1. change the `log` suffix to `txt`.
2. Don't load `ideolog` plugin;

**Comment by SkyeBeFreeman** [comment]
same.
Disable the Ideolog plugin for the time being.

**Comment by flyasky** [comment]
+1

**Comment by mfilippov** [comment]
Repro case:
<img width="603" alt="Screenshot 2024-03-04 at 12 14 37 PM" src="https://github.com/JetBrains/ideolog/assets/927432/f6f7e213-699b-4457-a4ce-bea231d8c778">

**Comment by mfilippov** [comment]
Fixed in 222.3.2.0
